### ISSUE TITLE
[Bug]: onDisconnection callback doesn't fire in Android

### ISSUE THREAD

**Issue by TheFe91** [apivideo/api.video-flutter-live-stream#38]
### Version

v1.1.1

### Which operating systems have you used?

- [X] Android
- [ ] iOS

### Environment that reproduces the issue

Samsung Galaxy A52 (Android 13)

### Is it reproducible in the example application?

Yes

### RTMP Server

MUX via rtmps (rtmps://global-live.mux.com:443/app)

### Reproduction steps

- Use the example app
- Set a MUX' stream key of your choice (or you can use this one: 6bf88cc5-b7f2-dc91-2814-33bf3d7b6abb)
- Start streaming
- Stream for some seconds (2-3 is fine)
- Stop streaming

### Expected result

The "Disconnected" snackbar shows

### Actual result

The "Disconnected" snackbar doesn't show

### Additional context

`flutter --version` output:

```
Flutter 3.10.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f468f3366c (5 weeks ago) • 2023-07-12 15:19:05 -0700
Engine • revision cdbeda788a
Tools • Dart 3.0.6 • DevTools 2.23.1
```

`flutter doctor -v` output:

```
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.3.1 22E772610a darwin-arm64, locale it-IT)
    • Flutter version 3.10.6 on channel stable at /Applications/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (5 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/alessandrodefendenti/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Users/alessandrodefendenti/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.10121639/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Users/alessandrodefendenti/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.10121639/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Users/alessandrodefendenti/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.9971841/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2023.1.3)
    • IntelliJ at /Users/alessandrodefendenti/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/231.9161.38/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2023.1.2)
    • IntelliJ at /Users/alessandrodefendenti/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/231.9011.34/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (4 available)
    • SM A525F (mobile)             • RZ8T50FAH4M               • android-arm64  • Android 13 (API 33)
    • iPhone di Alessandro (mobile) • 00008110-001558810EB8401E • ios            • iOS 16.5.1 20F75
    • macOS (desktop)               • macos                     • darwin-arm64   • macOS 13.3.1 22E772610a darwin-arm64
    • Chrome (web)                  • chrome                    • web-javascript • Google Chrome 115.0.5790.170

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

**Comment by ThibaultBee** [comment]
Hello,

Sorry for the delay, I have been pretty busy lately.
When you mean `Stop streaming` are you talking about calling `stopStream` API?

If this is the case, could you check that https://github.com/apivideo/api.video-flutter-live-stream/pull/41 fixes your issue?

**Comment by TheFe91** [comment]
Yeah, that probably is the solution! Waiting for it! Thank you!

**Comment by ThibaultBee** [comment]
It should be corrected in v1.1.3: https://github.com/apivideo/api.video-flutter-live-stream/releases/tag/v1.1.3
Please reopen this issue if you still face the problem
